### ISSUE TITLE
Blur hash addition

### ISSUE THREAD

**Issue by Silfalion** [appwrite/appwrite#1621]
## 🚀 Feature

Not sure how useful this would be but the idea is to add a blur hashing feature to appwrite, through the sdks or to the core itself.

As pointed by Meldiron, this could be kinda bloating for appwrite since the current solution to use a function to generate the hash then adding into a document that's paired to a pic in the storage worker works perfectly fine.

## Have you spent some time to check if this issue has been raised before?

Yes

### Have you read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/master/CODE_OF_CONDUCT.md)?

Yes

## Pitch

Honestly, not very sure it should but someone may have a special need or idea as to how it would be relevant.

The simplest case scenario is for loading data about a product and its picture. You load the data which contains the id to the pic in storage. The data would have the blur hash along with it to show a blur while we fetch the picture from appwrite.

And as pointed above this can easily be done with sdk librairies and cloud functions.

**Comment by eldadfux** [comment]
My main concern is that this might be too opinionated, if someone don't need this feature the overhead of getting the hash might be a bit too much. Maybe a dedicate endpoint or param as part of `/v1/storage/{fileId}/preview` can do the trick.

Anyway, I think it would be cool to have a cloud function that can do this thing on file creation.

**Comment by Silfalion** [comment]
Love the idea of integrating it to preview, it would fit nicely with the concept of editing photos, somehow. 

Cloud functions are plenty enough for this though, especially since the used packages are already maintained by third parties it would be one less hassle to maintain for Appwrite. 

On the other hand, the package wouldn't be that complex, I mean I don't see anything crazy that can be added to blur hash, is there?

**Comment by Meldiron** [comment]
I don't think this can be the official Appwrite Storage feature, as not all files are images, so not all files can have BlurHash. That would result in many files having `blurhash` value `null`. Also, creating blurhash can be pretty intensive work and might be a downside for projects that don't need it.

Previewing a file with a small resolution could do the trick in some cases, but I believe Blurhash is used to spend CPU power in order to reduce the required network. Considering 3G in the metro, even preview could take a few seconds to load, purely for HTTP connection establishment.

With that said, BlurHash exists to fill the content with blurry images instead of having skeleton loaders all over the place. This means you load BlurHash with content. In a real scenario, you would have a collection like `articles` with attribute like `coverFileId`.
My recommendation would be to have an Appwrite Function that would listen to an event of a new document created in this `articles` collection... This function would look at `coverFileId` attribute, get the file, do the calculation, and update the document, adding value to an attribute `blurHash`.
This way, if you load articles on front end, you automatically get BlurHash with your data. No need for a file request, no need for a preview request... That's the best usage of BlurHash that you can have. And of course, after rendering BlurHash, fire a request to get the file and replace BluHash as soon as you have the image file downloaded.

**Comment by eldadfux** [comment]
Stale issue, closing down.
